flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionTest.java

```diff
@@ -540,6 +551,46 @@ public void testSlotReleaseAtomicallyReleasesExecution() throws Exception {
 		});
 	}
 
+	/**
+	 * Tests that producedPartitions are registered synchronously under an asynchronous interface.
```

Comment:
```suggestion
	 * Tests that we reject incompleted futures returned by {@link ShuffleMaster#registerPartitionWithProducer...}.
```

Reply:
How about this:

Tests that asynchronous registrations (incomplete futures returned by {@link ShuffleMaster#registerPartitionWithProducer}) are rejected.

Reply:
How about this:

Tests that asynchronous registrations (incomplete futures returned by {@link ShuffleMaster#registerPartitionWithProducer}) are rejected.